src/integrations/terminal/ExecaTerminalProcess.ts

```diff
@@ -159,62 +283,75 @@ export class ExecaTerminalProcess extends BaseTerminalProcess {
-			// Try to kill using the subprocess object
-			if (this.subprocess) {
-				try {
-					this.subprocess.kill("SIGKILL")
-				} catch (e) {
-					console.warn(
-						`[ExecaTerminalProcess#abort] Failed to kill subprocess: ${e instanceof Error ? e.message : String(e)}`,
-					)
+		// Simplified process termination function: directly use process group kill (most reliable method)
+		const performKill = async () => {
+			if (!this.pid) {
+				// If no PID, only cleanup subprocess
+				if (this.subprocess) {
+					try {
+						if (typeof (this.subprocess as any).cancel === "function") {
+							;(this.subprocess as any).cancel()
+						}
+						this.subprocess = undefined
+					} catch (e) {
+						console.warn(
+							`[ExecaTerminalProcess#abort] Failed to cleanup subprocess: ${e instanceof Error ? e.message : String(e)}`,
+						)
+					}
 				}
+				return
 			}
 
-			// Kill the stored PID (which should be the actual command after our update)
-			if (this.pid) {
+			console.log(`[ExecaTerminalProcess#abort] Terminating process ${this.pid} and its process group`)
+
+			// Directly use process group kill (most reliable method, kills all related processes at once)
+			const killed = await this.forceKillProcessGroup(this.pid)
+
+			if (!killed) {
+				// If process group kill fails, try individual process kill as fallback
+				console.warn(
+					`[ExecaTerminalProcess#abort] Process group kill failed, trying individual process kill as fallback`,
+				)
+				await this.forceKillProcess(this.pid, true)
+			}
+
+			// Verify process is actually terminated
+			await new Promise((resolve) => setTimeout(resolve, 500))
+			try {
+				process.kill(this.pid, 0)
+				// Process is still running
+				console.error(
+					`[ExecaTerminalProcess#abort] Process ${this.pid} still running after all termination attempts`,
+				)
+			} catch (e) {
+				// Process has been terminated
+				console.log(`[ExecaTerminalProcess#abort] Process ${this.pid} successfully terminated`)
+			}
+
+			// Cleanup subprocess object
+			if (this.subprocess) {
 				try {
-					process.kill(this.pid, "SIGKILL")
+					// Try to cancel subprocess (if supported)
+					if (typeof (this.subprocess as any).cancel === "function") {
+						;(this.subprocess as any).cancel()
+					}
+					// Cleanup subprocess reference
+					this.subprocess = undefined
+					console.log(`[ExecaTerminalProcess#abort] Subprocess object cleaned up`)
 				} catch (e) {
 					console.warn(
-						`[ExecaTerminalProcess#abort] Failed to kill process ${this.pid}: ${e instanceof Error ? e.message : String(e)}`,
+						`[ExecaTerminalProcess#abort] Failed to cleanup subprocess: ${e instanceof Error ? e.message : String(e)}`,
 					)
 				}
 			}
 		}
 
 		// If PID update is in progress, wait for it before killing
 		if (this.pidUpdatePromise) {
-			this.pidUpdatePromise.then(performKill).catch(() => performKill())
+			this.pidUpdatePromise.then(() => performKill()).catch(() => performKill())
```

Comment:
to get more readable and stable way use that instead
`this.pidUpdatePromise.finally(performKill.bind(this)) `

finally is the better way than made twicely the same call in then or catch. finally is natvely support on promise system, please read the documentation to get more details about finally if you need some extra details.


btw the best way to reduce the code and keep it fonctional is to use finnaly @roomote